crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -456,24 +455,22 @@ async fn c8y_mapper_alarm_with_custom_fragment_mapping_to_c8y_json() {
 
     broker
         .publish_with_opts(
-            "tedge/alarms/major/temperature_alarm",
+            "tedge/alarms/major/custom_temperature_alarm",
             r#"{ "text": "Temperature high","time":"2023-01-25T18:41:14.776170774Z","customFragment": {"nested":{"value": "extra info"}} }"#,
             mqtt_channel::QoS::AtLeastOnce,
             true,
         )
         .await
         .unwrap();
 
-    let expected_msg = json!({"severity":"MAJOR","type":"temperature_alarm","time":"2023-01-25T18:41:14.776170774Z","text":"Temperature high","customFragment":{"nested":{"value":"extra info"}}});
-
-    while let Ok(Some(msg)) = messages.next().with_timeout(TEST_TIMEOUT_MS).await {
-        assert_json_include!(actual:serde_json::from_str::<serde_json::Value>(&msg).unwrap(), expected:expected_msg);
-    }
+    let expected_msg = r#"{"severity":"MAJOR","type":"custom_temperature_alarm","time":"2023-01-25T18:41:14.776170774Z","text":"Temperature high","customFragment":{"nested":{"value":"extra info"}}}"#;
 
+    // Expect converted temperature alarm message
+    mqtt_tests::assert_received_all_expected(&mut messages, TEST_TIMEOUT_MS, &[expected_msg]).await;
```

Comment:
This is still problematic as an assertion failure will prevent the cleanup logic below(clearing alarms) from executing. Not just in this test, but all alarm tests relying on this cleanup behaviour suffers from the same as you discovered yesterday.

Reply:
What I observed is that these tests run in serial and when the next test starts it will create a new broker instance, which will be a clean start and no persisted messages.

Reply:
So, we don't even need that "alarm cleanup" step in all these tests, huh? Fine then. You can remove those cleanup steps if it's useless anyway. But, up-to you.